src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java

```diff
@@ -82,6 +85,27 @@ public void placeInPortal(Entity entityIn, float rotationYaw) {
         }
     }
 
+    @Override
+    public void bridge$placeEntity(net.minecraft.world.World world, Entity entity, float yaw) {
+        boolean didPort;
+        if (entity instanceof EntityPlayerMP) {
+            this.placeInPortal(entity, yaw);
+            didPort = true;
+        } else {
+            if (((WorldServerBridge) this.world).bridge$getDimensionId() == 1) {
+                didPort = true;
+            } else {
+                didPort = this.placeInExistingPortal(entity, yaw);
+            }
+        }
+        
+        if (PhaseTracker.getInstance().getCurrentContext() instanceof InvokingTeleporterContext) {
```

Comment:
either replace with a `isTeleporting()` or make it a state based method that's overridden in the state, much like `isCapturing` or `isTicking`.

Reply:
That code comes from SpongeCommon but you don't have a bad point.